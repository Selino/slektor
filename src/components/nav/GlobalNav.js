import React from "react"
import { Nav, Navbar } from "react-bootstrap"
import { LinkContainer } from "react-router-bootstrap"
import logo from "../../logo.svg"

export default function globalNav() {
  return (
    <header className='App-header'>
      <Navbar fixed='top' bg='dark' variant='dark'>
        <Navbar.Brand href='#home'>
          <img
            alt='Slektor App Logo'
            src={logo}
            className='d-inline-block App-logo'
          />
          Slektor App
        </Navbar.Brand>
        <Navbar.Collapse id='basic-navbar-nav'>
          <Nav className='ml-auto'>
            <Nav.Item>
              <LinkContainer to='/booklist'>
                <Nav.Link>Books</Nav.Link>
              </LinkContainer>
            </Nav.Item>
            <Nav.Item>
              <LinkContainer to='/challenges'>
                <Nav.Link>Challenges</Nav.Link>
              </LinkContainer>
            </Nav.Item>
            <Nav.Item>
              <LinkContainer to='/create'>
                <Nav.Link>Create</Nav.Link>
              </LinkContainer>
            </Nav.Item>
            <Nav.Item>
              <LinkContainer to='/help'>
                <Nav.Link>Help</Nav.Link>
              </LinkContainer>
            </Nav.Item>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    </header>
  )
}
