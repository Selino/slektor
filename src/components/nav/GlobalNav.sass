@import "../../settings"

.GlobalNav
    color: #fff
    .app-logo
        color: #fff
        margin-right: .5rem
        vertical-align: middle

.bg-light
    background-color: $textColor