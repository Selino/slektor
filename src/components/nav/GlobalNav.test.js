import React from "react"
import GlobalNav from "./GlobalNav"
import { shallow } from "enzyme"

test("should render the global nav", () => {
  const wrapper = shallow(<GlobalNav />)
  expect(wrapper).toMatchSnapshot()
  // expect(wrapper.find(".App-logo")).toHaveLength(1)
  // expect(wrapper.find(".GlobalNav").text()).toContain("Slektor App")
  // expect(wrapper.find("#basic-navbar-nav")).toHaveLength(1)
})
