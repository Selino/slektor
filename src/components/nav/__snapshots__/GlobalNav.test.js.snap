// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the global nav 1`] = `
<header
  className="GlobalNav"
>
  <Navbar
    bg="dark"
    collapseOnSelect={false}
    expand={true}
    fixed="top"
    variant="dark"
  >
    <NavbarBrand
      href="#home"
    >
      <img
        alt="Slektor App Logo"
        className="d-inline-block App-logo"
        src={null}
      />
      Slektor App
    </NavbarBrand>
    <NavbarCollapse
      id="basic-navbar-nav"
    >
      <Nav
        className="ml-auto"
        fill={false}
        justify={false}
      >
        <NavItem>
          <withRouter(LinkContainer)
            to="/booklist"
          >
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
            >
              Books
            </NavLink>
          </withRouter(LinkContainer)>
        </NavItem>
        <NavItem>
          <withRouter(LinkContainer)
            to="/challenges"
          >
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
            >
              Challenges
            </NavLink>
          </withRouter(LinkContainer)>
        </NavItem>
        <NavItem>
          <withRouter(LinkContainer)
            to="/create"
          >
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
            >
              Create
            </NavLink>
          </withRouter(LinkContainer)>
        </NavItem>
        <NavItem>
          <withRouter(LinkContainer)
            to="/help"
          >
            <NavLink
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
            >
              Help
            </NavLink>
          </withRouter(LinkContainer)>
        </NavItem>
      </Nav>
    </NavbarCollapse>
  </Navbar>
</header>
`;
